python/tank/platform/engine.py

```diff
@@ -970,6 +970,8 @@ def register_command(self, name, callback, properties=None):
                 # we know the app for the existing item.
                 # so prefix with app name
                 prefix = existing_item["properties"].get("app").instance_name
+                if existing_item["properties"].get("group"):
+                    prefix = "%s:%s" % (prefix, existing_item["properties"]["group"])
```

Comment:
As if it was meant to be, It does not affect the display name! 😄 Here's a screen grab showing duplicate Maya 2016 and Maya 2016.5 launch commands. One set is not grouped, the other is grouped under "Maya". Sorry for the unfortunate placement of the drop down menu:
![4059_desktop_dup_commands](https://cloud.githubusercontent.com/assets/15038709/23322018/ea452680-fa96-11e6-84e9-42ea098051d6.png)

The "Maya Maya 2016" RECENT was launched from the Maya group, whereas the "Maya 2016" was launched from the command button. The duplicate "Maya Maya" is coming from the group name "Maya" for the BUTTON_ROLE and "Maya 2016" from the MENU_ROLE. It's not a by-product of this update.